Moving some settings to vimrc.custom

boombador opened the pull request:
Submitting this as a pull request because I think some of the changes might be controversial and I wanted to try submitting a pull request. Basically moving the mappings from the base vimrc as well as a couple of settings I found annoying.

boombador committed:
Move several settings and mappings to vimrc.local

I think it makes sense to move more settings into the vimrc.local file
since it's still available but the basic file under version control
stays unobjectionable. I'm open to debate on these but here is my
reasonning for each:

- the foldmethod line causes random files to open up folded for me
- nowrap makes it so you can't see everything on the screen at once
- hidden causes problems if you try to open files in new vim instances
  without fully closing out of your original vim session
- it looks like mappings depend on the leader definition at the time the
  mapping is made so it seemed to make sense to move the mapleader line
 and all the mappings that might depend on it together into the local
 file

boombador committed:
Fixing shebang on install script

/usr/env didn't work for me, does this work on other systems? Will my
suggestion of /usr/bin/env break it elsewhere?

ctaylo21 commented on vimrc.custom:
```diff
@@ -19,22 +42,68 @@ if has("gui_running")
+
+" using comma allows this though
 " For quick save/quit
 nnoremap ; :
```
I would remove comments based on a conditional setting. Instead of having "If you set leader to this then you can use this" maybe put the quick save option for leader = ; right underneath the commented out part where you can set leader to ; that way it's not confusing.

boombador replied:
My line setting the leader to ";" is commented out so it won't conflict, but per our conversation I can remove the inactive mapleader declaration to reduce confusion.

ctaylo21 commented on vimrc.custom:
```diff
@@ -19,22 +42,68 @@ if has("gui_running")
 nnoremap ; :
 
+" For quick save/quit
+nnoremap <leader><space> :
```
Is this really any faster than the default of ":"?

ctaylo21 commented on vimrc.custom:
```diff
@@ -19,22 +42,68 @@ if has("gui_running")
+nmap <leader>n :NERDTree<CR>
+
+" Opens current file heiarchy in Nerdtree
+nmap <leader>nf :NERDTreeFind<CR>
```
Was there a reason this was changed from <leader>f to <leader>nf?

boombador replied:
Oops, accidentally included a change from my vimrc.local. Will undo that mapping change in my next commit.

ctaylo21 commented on vimrc.custom:
```diff
@@ -19,22 +42,68 @@ if has("gui_running")
-" Allows you to save files you opened without write permissions
-cmap w!! w !sudo tee %
+" Hides buffers instead of closing them
+set hidden
```
I would argue that this belongs in the default vimrc. Without this vim will not let you have hidden buffers with local changes. That means if you switch back and forth between files you will have to save each time before leaving the window.

boombador committed:
Implementing feedback on vimrc.custom file

- change back to agreed mapping of <leader>f for NERDTreeFind
- remove confusing comments about alternate mapleaders
- remove my weird <leader><space> mapping from the vimrc.custom file
- didn't move the "hidden" option back to the main file because it
  proved contentious
